### PR TITLE
Tweak styles and incorporate "Add a Plot" into lefthand menu.

### PR DESCRIPTION
**Pull request by adnauseum**
Bob's most excellent PR (https://github.com/VEuPathDB/web-eda/pull/1690) brought in the ability for users to add a new visualization to the standalone map app. Hence the small _net_ diff.
 
This PR Piggybacks incorporates most of those components into the side navigation. 

A subsequent PR will break out the draggable panels! 🤘 

**Nota bene**:
- This PR includes Bob's addendum PR: #22 
- We should think about how we'll style the visualizations grid which is optimal for a lot of space.

<details>
<summary>See quick demo of the draggable panels here!</summary>

https://user-images.githubusercontent.com/24446573/228222505-5ebf7016-0f21-4ad7-98fd-2db7e1d6d163.mp4

</details>

**Before**:
Screenshot 2023-03-27 at 11 15 00 AM
Screenshot 2023-03-27 at 11 14 51 AM

**After**:
Screenshot 2023-03-28 at 3 41 23 PM

image
Screenshot 2023-03-27 at 11 14 22 AM

### CODE REVIEW

File: packages/libs/eda/src/lib/map/analysis/FloatingVizManagement.tsx [hunk 1]
```diff
@@ -201,27 +202,36 @@ export default function FloatingVizManagement({
 
   return (
     <>
-      <FloatingDiv
-        style={{
-          top: 200,
-          left: 200,
-        }}
-      >
+      <div>
```
Comment (adnauseum) [inline packages/libs/eda/src/lib/map/analysis/FloatingVizManagement.tsx:205]: **Note to benevolent reviewer(s)**: the changes around here move the "Add a Plot" stuff into the side navigation. More than that, they also add some styling/verbiage tweaks discussed on [2023-03-23's Map UX meeting](https://veupathdb.atlassian.net/wiki/spaces/WG/pages/227573761/2023-03-23+Map+UX).

Screenshot 2023-03-27 at 2 45 14 PM

File: packages/libs/eda/src/lib/map/analysis/MapAnalysis.tsx [hunk 2]
```diff
@@ -34,15 +34,25 @@ import {
...
 import { InputVariables } from '../../core/components/visualizations/InputVariables';
 import { useToggleStarredVariable } from '../../core/hooks/starredVariables';
 import { filtersFromBoundingBox } from '../../core/utils/visualization';
+import {
+  BarChartSharp,
+  EditLocation,
+  InfoSharp,
+  Save,
+  Share,
+} from '@material-ui/icons';
```
Comment (adnauseum) [inline packages/libs/eda/src/lib/map/analysis/MapAnalysis.tsx:55]: **Note to benevolent reviewer(s)**: these were added to flesh out the side navigation as per the discussed on [2023-03-23's Map UX meeting](https://veupathdb.atlassian.net/wiki/spaces/WG/pages/227573761/2023-03-23+Map+UX).

image

File: packages/libs/eda/src/lib/map/analysis/MapAnalysis.tsx [hunk 3]
```diff
@@ -54,6 +64,47 @@ interface Props {
   siteInformationProps: SiteInformationProps;
 }
 
+type App = {
```
Comment (adnauseum) [inline packages/libs/eda/src/lib/map/analysis/MapAnalysis.tsx:67]: **Question to benevolent reviewer(s)**: I wanted to get the type of `app`. This is copy/pasted from VS Code. What's the better way of doing this?

File: packages/libs/eda/src/lib/map/analysis/MapAnalysis.tsx [hunk 4]
```diff
@@ -257,44 +308,88 @@ function MapAnalysisImpl(props: Props & CompleteAppState) {
...
+          }}
+        >
+          <span style={iconStyles} aria-hidden>
+            {icon}
+          </span>
+          <span style={labelStyles}>{labelText}</span>
+        </button>
+      );
+    }
+  );
```
Comment (adnauseum) [inline packages/libs/eda/src/lib/map/analysis/MapAnalysis.tsx:392]: **Note to benevolent reviewer(s)**: this and following lines are changes made to the configuration and rendering of the lefthand side navigation items.

**Review by bobular — CHANGES_REQUESTED** [review]
Looks great.

At the moment, when a menu tab is activated, its component is rendered "afresh" via the `renderWithApp` function.

In order to get the "activeViz" inappropriately named `FullScreenVisualization` to persist when the menu item is closed, we're going to have to do some restructuring.

OR should we render all the menu tabs but only have the active one visible?

**Comment by adnauseum** [comment]
> Looks great.
> 
> At the moment, when a menu tab is activated, its component is rendered "afresh" via the renderWithApp function.
> 
> In order to get the "activeViz" inappropriately named FullScreenVisualization to persist when the menu item is closed, we're going to have to do some restructuring (specifically: move the "activeViz" component out of this component into the parent MapAnalysis - actually this maybe isn't as problematic as I thought)
> 
> OR should we render all the menu tabs but only have the active one visible?

So this would be a scenario illustrating your thoughts:
* User clicks **Plot**.
* User selects a histogram and enters the name "My histo" without saving.
* User goes back to the side nav and pressed **Filter**.
* User does some awesome filtering! 👏 
* User clicks **Plot** again.
* _User should see their partially configured histogram with the name "My histo" in the editable name field_.

Right now the user returning to **Plot** would see everything reset.

File: packages/libs/eda/src/lib/map/analysis/MapAnalysis.tsx (hunk 4 above)
Comment (adnauseum) [reply packages/libs/eda/src/lib/map/analysis/MapAnalysis.tsx]: Resolved in: https://github.com/VEuPathDB/web-monorepo/pull/8/commits/c853e0b19b033d59d553093b24d545c941190ada

File: packages/libs/eda/src/lib/map/analysis/MapAnalysis.tsx (hunk 3 above)
Comment (adnauseum) [reply packages/libs/eda/src/lib/map/analysis/MapAnalysis.tsx]: Resolved by https://github.com/VEuPathDB/web-monorepo/pull/8/commits/73c5b41c7d1cfd0bed13bd0a782600c47a7a5fb2

**Review by bobular — APPROVED** [review]

**Comment by bobular** [comment]
Re-requesting my own review because it looks like we need to leave this in a semi-attractive functional state before merging (so that I and maybe others can work on other menu items).

I suggest addressing only the following:

- [ ] all the new-plot icons fit on screen (or see [here](https://epvb.slack.com/archives/CMQ8K9XRS/p1680117273090959) for an alternative with fewer clicks)
- [ ] the floating plot in a draggable
- [ ] the activate floating plot highlighted in the list of plots
- [ ] make a new ticket for fine-tuned styling

**Pull request merged**